### DISCUSSION TITLE
Import listens via API

### DISCUSSION THREAD

**Discussion by SimonHova** [ampache/ampache#3430] [Ideas]
For some time, I've been backing up my ratings to MusicBrainz, and it's been working very well. So much that when I had a database crash, I was able to recover with very little data lost, because the most critical piece of information, my ratings, was being backed up. I'm even more proud of the fact that it's 100% API driven, and I've eliminated any reliance on the database. However, I don't yet have any way to restore my scrobbled listens via the API. I know that it's a feature that will likely never be used by anyone other than me, but can we put it on the roadmap for the future?

**Comment by lachlan-00** [comment]
You can totally already do that! I imported my library from last fm using a csv and python. (old and terrible code here https://github.com/lachlan-00/ampache-scripts/blob/master/scrobble_to_ampache_from_file.py)

Record play for when you know the id of the object
https://ampache.org/api/api-json-methods#record_play

Scrobble to scrobble the item based on name, etc.
https://ampache.org/api/api-json-methods#scrobble

When you have the date there should be no collisions and everything acts as normal

**Reply by stebe** [reply]
What did you use to scrape your last.fm scrobbles? 

Happy New Year

**Reply by lachlan-00** [reply]
https://github.com/lachlan-00/lastscrape-gui/blob/master/lastexport3.py i don't know it it still works but i was able to export the whole lot from 2005.

**Reply by lachlan-00** [reply]
https://github.com/encukou/lastscrape-gui the original project does seem to have activity happening again though
